src/datasources/postgres/ProposalDataSource.ts

```
@@ -335,9 +355,41 @@ export default class PostgresProposalDataSource implements ProposalDataSource {
             .andWhere('answers.question_id', questionFilter.questionId)
             .modify(questionFilterQuery, questionFilter);
         }
+        if (
+          searchText !== '' &&
+          searchText !== null &&
+          searchText !== undefined
+        ) {
+          searchText = `%${searchText}%`;
+
+          query
+            .whereRaw('proposal_table_view.proposal_id ILIKE ?', searchText)
+            .orWhereRaw('proposal_table_view.title ILIKE ?', searchText)
+            .orWhereRaw('proposal_status_name ILIKE ?', searchText)
+            .orWhereRaw('instrument_name ILIKE ?', searchText);
+        }
+
+        if (sortField && sortDirection) {
+          sortField = fieldMap[sortField];
```

What happens if the `sortField` is not found here? I think we should throw a pre-emptive exception.

---

I am not sure how that could happen unless querying directly. Do you mean to check the field and to throw a different exception before putting it into the map?

---

As this is also going to be a developer facing API, querying the GraphQL API directly is possible.

If the `fieldMap[sortField]` is the "not found" value (I don't know if it's `null` or `undefined`) then we should throw an exception.